rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -364,12 +366,13 @@ def add_acquisition(self, acquisition: AssetAcquisitionEvent) -> None:
             self._acquisitions_heap,
             AssetAcquisitionHeapElement(self._count, acquisition),
         )
-        self.current_average_cost_basis += (acquisition.rate * acquisition.amount)
+        self.total_value += acquisition.rate * acquisition.amount
         self.remaining_amount += acquisition.amount
         self._count += 1
 
     def consume_result(self, used_amount: FVal) -> None:
         """Same as its parent function but also deducts `used_amount` from `remaining_amount`."""
+        self.total_value -= used_amount * self.total_value / self.remaining_amount
```

Comment:
There are four places where `consume_result` is called. In all of them it is checked that `remaining_amount_ is not zero. Otherwise a missing acquisition is thrown
![image](https://user-images.githubusercontent.com/63492346/217041809-4ef3cf57-f91b-45bd-9e51-64438519d2c7.png)

Reply:
then you could write an assert in the code (since it will get removed in the packaged binary) and add a comment to the docstring saying that this condition is assumed

Reply:
I don't think that we add assertions for such things as they depend solely on us (developers).

Also, for example, here, there is no such assertion https://github.com/rotki/rotki/blob/e19c67dab191424d07ae40f2f86235509a2b54c3/rotkehlchen/accounting/cost_basis/base.py#L158

Reply:
fair, I guess that if it's broken we would see the divisionbyzeroerror during testing but at the condition to the docstring at least